### PR TITLE
[WM1533][WM-1555] New GET /run_sets API and updated GET /runs API to optionally return Runs for given Run Set

### PR DESCRIPTION
**Pull request by kpierre13**
This PR:
- adds a new API GET /run_sets that returns details of all Run Sets in a list form
- updates API GET /runs to optionally take in a run_set_id parameter to return Runs for that particular Run Set

Closes https://broadworkbench.atlassian.net/browse/WM-1533 and https://broadworkbench.atlassian.net/browse/WM-1555

### CODE REVIEW

**Comment by salonishah11** [comment]
I am not sure how to get rid of [this code smell](https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=81&id=DataBiosphere_terra-batch-analysis&open=AYSMCjkukVmPCwkFTYVh) as we follow the same pattern everywhere 🤔

**Review by salonishah11 — APPROVED** [review]
Katrina's changes LGTM 👍

File: service/src/main/java/bio/terra/cbas/dao/RunDao.java [hunk 1]
```diff
@@ -31,11 +32,19 @@ public int createRun(Run run) {
         new EnumAwareBeanPropertySqlParameterSource(run));
   }
 
-  public List<Run> getRuns() {
+  public List<Run> getRuns(String runSetId) {
+    String whereClause = runSetId == null ? "" : " WHERE run.run_set_id = :runSetId";
```
Comment (mspector) [inline service/src/main/java/bio/terra/cbas/dao/RunDao.java:36]: I didn't realize we can do this in Java! Nice!

**Review by mspector — APPROVED** [review]
I'm approving this because the code smell thing is really no big deal, but I admit it does make me pause ever so slightly. Defining those strings as variables instead of duplicating them does help eliminate a certain class of error, perhaps at the expense of verbosity.

I'd probably make that change unless it looked really awful or seemed ridiculous, but I won't let it stop me from approving

**Comment by salonishah11** [comment]
Two separate PRs will be created for the two different tickets. Closing this PR.

**Pull request closed without merging**